src/cdk-experimental/listbox/listbox.ts

```diff
@@ -0,0 +1,138 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.dev/license
+ */
+
+import {
+  computed,
+  contentChildren,
+  Directive,
+  ElementRef,
+  inject,
+  input,
+  model,
+  OnDestroy,
+  signal,
+} from '@angular/core';
+import {OptionPattern} from '../ui-patterns/listbox/option';
+import {ListboxInputs, ListboxPattern} from '../ui-patterns/listbox/listbox';
+import {Directionality} from '@angular/cdk/bidi';
+import {startWith, takeUntil} from 'rxjs/operators';
+import {Subject} from 'rxjs';
+
+@Directive({
```

Comment:
Even as experimental, we should add a JsDoc description for this class, including what it's responsible for and  how its intended to be used.